### PR TITLE
feat([sc-16295]): approve icrc1_transfer identitykit screen

### PR DESCRIPTION
**Pull request by olekid**

### CODE REVIEW

File: apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  const actor = actorService.getActor<any>(
+    message.data.params.canisterId,
+    icrc1and2IDL,
+    agent as never,
+  )
+
+  const { owner, subaccount } = decodeIcrcAccount(message.data.params.sender)
+
+  const [balance, symbol, decimals, fee] = await Promise.all([
+    actor.icrc1_balance_of({
```
Comment (olsemeno) [inline apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts:32]: why not icrc1_metadata?

File: apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }),
+    actor.icrc1_symbol(),
+    actor.icrc1_decimals(),
+    actor.icrc1_fee(),
+  ])
+
+  const [requestParams] = JSON.parse(args)
+
+  const amount = Number(requestParams.amount)
+  const isInsufficientBalance = amount + Number(fee) > Number(balance)
```
Comment (olsemeno) [inline apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts:44]: are you sure that number works fine with 16+ decimails?

File: apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  args: any,
+): Promise<TransferMetadata> => {
+  const delegation = new AnonymousIdentity()
+
+  const agent: Agent = new HttpAgent({
+    host: IC_HOSTNAME,
+    identity: delegation,
+  })
+
+  const actor = actorService.getActor<any>(
```
Comment (olsemeno) [inline apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts:23]: you can pass Service instead of any

**Review by olsemeno — APPROVED** [review]

File: apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts (hunk 3 above)
Comment (olekid) [reply apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts]: done

File: apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts (hunk 2 above)
Comment (olekid) [reply apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts]: changed to strings everywhere, main component which renders all numbers in app changed to accept number or string and convert to bignumbers under the hood for backward compatibility

File: apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts (hunk 1 above)
Comment (olekid) [reply apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts:33]: makes sense, actually was copied from other methods, but i personally don't like return type of metadata, its [[,symbol], [, decimals], [, [fee]], i am not sure it's required

**Pull request merged**